src/dmd/libomf.d

```diff
@@ -167,29 +167,28 @@ final class LibOMF : Library
 
     /*****************************************************************************/
 
-    void addSymbol(OmfObjModule* om, const(char)* name, int pickAny = 0)
+    void addSymbol(OmfObjModule* om, const(char)[] name, int pickAny = 0)
     {
         static if (LOG)
         {
-            printf("LibOMF::addSymbol(%s, %s, %d)\n", om.name.ptr, name, pickAny);
+            printf("LibOMF::addSymbol(%s, %s, %d)\n", om.name.ptr, name.ptr, pickAny);
         }
-        const namelen = strlen(name);
-        StringValue* s = tab.insert(name, namelen, null);
+        StringValue* s = tab.insert(name, null);
         if (!s)
         {
             // already in table
             if (!pickAny)
             {
-                const s2 = tab.lookup(name, namelen);
+                const s2 = tab.lookup(name);
                 assert(s2);
                 const os = cast(const(OmfObjSymbol)*)s2.ptrvalue;
-                error("multiple definition of %s: %s and %s: %s", om.name.ptr, name, os.om.name.ptr, os.name);
+                error("multiple definition of %s: %s and %s: %s", om.name.ptr, name.ptr, os.om.name.ptr, os.name);
```

Comment:
Even more important here. Should have been done for `om.name.ptr` as well. FYI @thewilsonator